warehouse/internal/repo/load.go

```diff
@@ -0,0 +1,180 @@
+package repo
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/rudderlabs/rudder-server/utils/timeutil"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+const (
+	loadTableName    = warehouseutils.WarehouseLoadFilesTable
+	loadTableColumns = `
+		id,
+		staging_file_id, 
+		location, 
+		source_id, 
+		destination_id, 
+		destination_type, 
+		table_name, 
+		total_events, 
+		metadata
+`
+)
+
+type LoadFiles struct {
+	DB  *sql.DB
+	Now func() time.Time
+
+	once sync.Once
+}
+
+func (repo *LoadFiles) init() {
+	repo.once.Do(func() {
+		if repo.Now == nil {
+			repo.Now = timeutil.Now
+		}
+	})
+}
+
+// DeleteByStagingFiles deletes load files associated with stagingFileIDs.
+func (repo *LoadFiles) DeleteByStagingFiles(ctx context.Context, stagingFileIDs []int64) error {
+	repo.init()
+
+	sqlStatement := `
+		DELETE FROM
+		  ` + loadTableName + `
+		WHERE
+		  staging_file_id = ANY($1);`
```

Comment:
```suggestion
		  staging_file_id IN ($1);`
```

Reply:
Due to syntactic restrictions, `IN` can not replace `= ANY`. 

We are passing a single prepared param, and for each value, we use `pq.Array`, which essentially results in `'{1,2,3}'` (assuming an `[]int{1,2,3}`).

If we were to use `IN` correctly with prepared params we would have to use multiple placeholder, one for each item in the array `IN ($1, $2, $3, $4 ....)`, which would unnecessarily complicate things. 

If there is a good performance reason to use `IN` we can consider the extra complexity, but from my short research I couldn't find a conclusive answer.

A couple of refs: 
* https://stackoverflow.com/a/31192557
* https://stackoverflow.com/a/34627688
* https://github.com/lib/pq/blob/master/array.go#L539